mkdocs/production/auth/custom-auth.md

```diff
@@ -1 +1,142 @@
+                extra=user_data.get("extra") # Optional data to be passed to the Cat
+            )
+        else:
+            return None  # If the server responds with an error or doesn't respond at all
+        
+        
+    async def authorize_user_from_key(self, protocol: str, user_id: str, api_key: str, auth_resource, auth_permission):
+        # Optional: Handle API key authentication, if applicable
+        return None
+
+```
+
+### Create the Custom Handler Config 
+
+Create a new `AuthHandlerConfig` class that inherits from `BaseAuthHandlerConfig`.
+This will be the configuration class that contains all the necessary settings to initialize your custom handler.
+
+```python
+from cat.factory.auth_handler import AuthHandlerConfig
+from typing import Type
+from pydantic import ConfigDict, Field
+
+class CustomAuthHandlerConfig(AuthHandlerConfig):
+    _pyclass: Type = CustomAuthHandler
+
+    server_url: str = Field(..., description="The URL of the identity provider")
+
+    model_config = ConfigDict(
+        json_schema_extra={
+            "humanReadableName": "Custom Auth Handler",
+            "description": "Delegate auth to a custom identity provider."
+        }
+    )
+
+```
+
+### Register the Custom Handler
+
+Register the new handler with the `factory_auth_handlers` hook.
+
+```python
+from cat.mad_hatter.decorators import hook
+from typing import List
+
+@hook(priority=0)
+def factory_allowed_auth_handlers(allowed: List[AuthHandlerConfig], cat) -> List:
+    allowed.append(CustomAuthHandlerConfig)
+    return allowed
+
+```
+
+### Activate the Custom Handler
+
+Activate the new handler using the proper endpoint in your Cat installation.
+
+```bash
+curl --location --request PUT 'http://{your_cat_instance}/auth_handler/settings/CustomAuthHandlerConfig' \
+--header 'Content-Type: application/json' \
+--data '{
+   "server_url": {your_idp_url},
+}'
+```
+
+That's it! Now every incoming request will be authenticated using the custom auth handler.
+
+## Key Components and Concepts
+
+### BaseAuthHandler
+
+`BaseAuthHandler` is the class that you will extend to implement your custom auth flow. This is an abstract class that
+forces you to implement two basic methods: `authorize_user_from_jwt` and `authorize_user_from_key`. You'll write your
+own auth logic in there inserting, for example, credentials validation request to external identity providers.
+
+- `authorize_user_from_jwt`: Validates and processes incoming JWT tokens, extracting user information and 
+- mapping it into Cheshire Cat’s `AuthUserInfo` structure. 
+- `authorize_user_from_key`: Allows validation of requests based on an API key, usually used for machine to machine communication.
+
+### AuthHandlerConfig
+
+The `AuthHandlerConfig` is the configuration class from which your own will inherit to create your own configuration
+class for your auth handler. It includes all the necessary settings to initialize your custom handler.
```

Comment:
I think the first sentence needs to be reworked a little, `is the configuration class from which your own will inherit to create your own configuration` sounds confusing